pymeasure/instruments/lecroy/lecroyT3DSO1204.py

```diff
@@ -0,0 +1,1324 @@
+    output = (value[0], sanitize_source(value[1]), sanitize_source(value[2]))
+    if output[1][0] > output[2][0]:
+        raise ValueError(f'First channel number {output[1]} must be <= than second one {output[2]}')
+    for i in range(3):
+        strict_discrete_set(output[i], values=values[i])
+    return output
+
+
+def _set_measure_parameter_validator(value, values):
+    """ Validate the input of the set_measure_parameter property
+    :param value: input parameters as a 2-element tuple
+    :param values: allowed space for each parameter """
+    if not isinstance(value, tuple):
+        raise ValueError('Input value {} of trigger_select should be a tuple'.format(value))
+    if len(value) != 2:
+        raise ValueError('Number of parameters {} different from 2'.format(len(value)))
+    output = (value[0], sanitize_source(value[1]))
+    for i in range(2):
+        strict_discrete_set(output[i], values=values[i])
+    return output
+
+
+def _intensity_validator(value, values):
+    """ Validate the input of the intensity property (grid intensity and trace intensity)
+    :param value: input parameters as a 2-element tuple
+    :param values: allowed space for each parameter """
+    if not isinstance(value, tuple):
+        raise ValueError('Input value {} of trigger_select should be a tuple'.format(value))
+    if len(value) != 2:
+        raise ValueError('Number of parameters {} different from 2'.format(len(value)))
+    for i in range(2):
+        strict_discrete_range(value=value[i], values=values[i], step=1)
+    return value
+
+
+class _ChunkResizer:
+    """ The only purpose of this class is to resize the chunk size of the instrument adapter.
+    This is necessary when reading a big chunk of data from the oscilloscope like image dumps and
+    waveforms.
+    Note.
+    Only if the new chunk size is bigger than the current chunk size, it is resized. """
+
+    def __init__(self, adapter, chunk_size):
+        """ Just initialize the object attributes.
+        :param: adapter of the instrument. This is usually accessed through the
+        Instrument::adapter attribute.
+        :chunk_size: new chunk size (int).
+        """
+        self.adapter = adapter
+        self.old_chunk_size = None
+        self.new_chunk_size = int(chunk_size) if chunk_size else 0
+
+    def __enter__(self):
+        """ Only resize the chunk size if the adapter support this feature"""
+        if (self.adapter.connection is not None
+                and hasattr(self.adapter.connection, "chunk_size")):
+            if self.new_chunk_size > self.adapter.connection.chunk_size:
+                self.old_chunk_size = self.adapter.connection.chunk_size
+                self.adapter.connection.chunk_size = self.new_chunk_size
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        if self.old_chunk_size is not None:
+            self.adapter.connection.chunk_size = self.old_chunk_size
+
+
+# noinspection DuplicatedCode
+class Channel:
```

Comment:
Channel implementation has arrived, such that you may inherit from "pymeasure.instruments.Channel".

Reply:
That is very good news. But would it be possible to open a new pull request after we merge this one, with the new Channel implementation?

Reply:
In the new pull request, I would apply all suggestion you made about the Channel class.

Reply:
Ok we can do it afterwards.
For now: would you mind to rename your channels from `ch1` to `ch_1`? That way they will adhere to the new standard, which we try to establish?

Reply:
I'm not sure what it would accomplish to open a separate pull request? In the end, all the comments/feedback from here would have to be collected and re-checked in the new PR?

Reply:
Psychology: It's some time this PR is open and probably Last Stardust wants to finish the implementation of this large PR. At least that would be my desire. 
Reading how that new channels stuff works, might feel like a major setback. He does not know what he has to change. 
If the channels are named `ch_1`, the usage won't change, just the implementation.

On the other, yes implementing channels is only minor work (I basically gave already all the necessary code) and it is better to do it in one go, than to have an instrument to be changed in the near future.

@LastStarDust, what do you think, if I gave you all the code suggestions and you change the channel names from `ch1` to `ch_1` and run the tests?
Thanks to the tests, we can verify easily, that nothing changed regarding usage.

Reply:
Ok let's do this now. I have renamed chX to ch_X.
Yes the reason was kind of psychological. I am using pymeasure at work to automate acceptance tests. Since the development of the oscilloscope driver is dragging on for a long time, my superior is pushing me to stop fussing over it and go on with other tasks.

Reply:
Thanks for your work.